### PR TITLE
Item 9204: Sample type parent import alias inclusion in download template and display in details panel

### PR DESCRIPTION
**Pull request by cnathe**
#### Rationale
Sample type definitions allow for parent import aliases to be set to make it easier for users to map columns in the sample import files to the parent samples / sources. This set of PRs exposes those parent import aliases in the sample type details panel and download template column headers.

#### Related Pull Requests
* https://github.com/LabKey/labkey-ui-components/pull/581
* https://github.com/LabKey/platform/pull/2458
* https://github.com/LabKey/sampleManagement/pull/628
* https://github.com/LabKey/biologics/pull/941

#### Changes
* Add FileUploadPanel helper to download template file

### CODE REVIEW

**Review by labkey-danield — APPROVED** [review]

**Comment by cnathe** [comment]
@labkey-danield I decided to go ahead and add a quick test case to the existing SampleTypeParentColumnTest for this story. Would you mind re-reviewing this PR?

File: src/org/labkey/test/pages/ImportDataPage.java [hunk 1]
```diff
@@ -118,6 +123,19 @@ public void cancel()
...
+    {
+        return doAndWaitForDownload(()->elementCache().getDownloadTemplateButton().click());
+    }
+
+    public List<String> getTemplateColumnHeaders() throws IOException
+    {
+        File template = downloadTemplate();
+        Workbook workbook = ExcelHelper.create(template);
+        Sheet sheet = workbook.getSheetAt(0);
+        return ExcelHelper.getRowData(sheet, 0);
```
Comment (labkey-danield) [inline src/org/labkey/test/pages/ImportDataPage.java]: I would have this function return just the file and put the ExcelHelper and Workbook code in the test case.

File: src/org/labkey/test/tests/SampleTypeParentColumnTest.java [hunk 2]
```diff
@@ -301,9 +302,24 @@ public void testValidAliasNames()
...
+        ImportDataPage importDataPage = sampleHelper.getSamplesDataRegionTable().clickImportBulkData();
+        try
+        {
+            List<String> actualHeaderValues = importDataPage.getTemplateColumnHeaders();
+            Assert.assertTrue("Download template file does not contain import alias", actualHeaderValues.contains(PARENT_COLUMN_1));
+        }
+        catch (IOException rethrow)
+        {
+            throw new RuntimeException(rethrow);
+        }
```
Comment (labkey-danield) [inline src/org/labkey/test/tests/SampleTypeParentColumnTest.java]: More or less just a style comment, but I usually don't catch and rethrow exceptions in the test case I just add a throws IOException to the to the function declaration:
`public void testValidAliasNames() throws IOException`

**Review by labkey-danield — APPROVED** [review]
Just a couple of comments.

File: src/org/labkey/test/pages/ImportDataPage.java (hunk 1 above)
Comment (cnathe) [reply src/org/labkey/test/pages/ImportDataPage.java]: done

File: src/org/labkey/test/tests/SampleTypeParentColumnTest.java (hunk 2 above)
Comment (cnathe) [reply src/org/labkey/test/tests/SampleTypeParentColumnTest.java]: that works for me. changed.

**Pull request merged**